test/unit/test_iostream.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#pragma once
+
+#include "core/iostream.h"
+/**
+ * A testing stream implementation. Doesn't write anything anywhere, but is
+ * useful for testing coio behaviour when waiting for stream to become readable
+ * or writeable.
+ */
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+struct test_stream_ctx {
+	/** How many bytes can be written before the stream would block. */
+	size_t avail_wr;
+	/** How many bytes can be read before the stream would block. */
+	size_t avail_rd;
+	/**
+	 * A file descriptor which is never ready for reads. Points to rfds[0],
+	 * the reading end of a pipe which is never written to.
+	 */
+	int rfd;
+	/** A pipe used to simulate a fd never ready for reads. */
+	int rfds[2];
+	/**
+	 * A file descriptor which is never ready for writes. Points to
+	 * wfds[1], the writing end of a filled up pipe.
+	 */
+	int wfd;
+	/** A pipe to simulate a fd never ready for writes. */
+	int wfds[2];
+};
+
+struct test_stream {
+	struct iostream io;
+	struct test_stream_ctx ctx;
+};
```

Comment:
`iostream` isn't supposed to be inherited. There's `void *ctx` in `struct iostream`. Anyway, as I said above, I don't see a reason to create a new iostream kind to test this fix - AFAIU you can simply wrap a pipe fd in the plain iostream.

Reply:
I just didn't want to bother with ctx allocation. Anyway, that doesn't matter after the rework.